### PR TITLE
Improve Build, and Build Documentation

### PR DESCRIPTION
**Pull request by jamestalmage**
This covers a myriad of build related issues.

See each commit message for more details.

### CODE REVIEW

**Comment by jamestalmage** [comment]
The image link is broken because the image has not been merged into master yet. It will work once merged.

**Comment by jamestalmage** [comment]
I think this now gets us close enough to close #17

File: index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function FirebaseServer(port, name, data) {
 	this.Firebase = firebaseCopy();
 	this.name = name || 'mock.firebase.server';
 	this.Firebase.goOffline();
-	this.baseRef = new this.Firebase('ws://fakeserver.firebaseio.com');
+	this.baseRef = new this.Firebase('ws://fakeserver.firebaseio.test');
```
Comment (jamestalmage) [inline index.js]: This needs to make it in regardless.

Ending in `.com` causes error messages to be logged to the console. Running the new `watch` target leaves the process open long enough that a LOT more of them get logged, and it's really annoying.

File: package.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
     "firebase-token-generator": "^2.0.0",
     "istanbul": "0.4.0",
     "mocha": "2.3.3",
-    "mockery": "1.4.0"
+    "proxyquire": "^1.7.3"
```
Comment (jamestalmage) [inline package.json]: I don't know why, but mockery did not like `mocha --watch`, switching to proxyquire solved the problem.

File: package.json [hunk 3]
```diff
@@ -4,10 +4,10 @@
...
   "main": "index.js",
   "scripts": {
-    "test": "npm run lint && npm run cover && npm run test-es5",
-    "cover": "istanbul cover ./node_modules/mocha/bin/_mocha -- -r test/mocha-babel.js test/*.spec.js",
-    "test-es5": "mocha test/es5-only-test.js",
-    "debug": "mocha -r test/mocha-babel.js test/*.spec.js",
+    "test": "npm run lint && npm run cover",
+    "cover": "istanbul cover ./node_modules/mocha/bin/_mocha",
+    "debug": "mocha",
+    "watch": "mocha -w",
```
Comment (jamestalmage) [inline package.json]: Using `mocha.opts` made all these scripts a lot simpler. Not everyone wants to install `mocha` globally, so the `debug` and `watch` scripts are still valuable.

File: test/mocha-babel.js [hunk 4]
```diff
@@ -10,5 +10,7 @@ function areGeneratorsSupported() {
 }
 
 if (!areGeneratorsSupported()) {
-	require('babel/register');
+	require('babel/register')({
+		ignore: /index\.js|lib\/|node_modules/
+	});
```
Comment (jamestalmage) [inline test/mocha-babel.js]: The separate ES5 sanity check is no longer necessary. This simply prevents babel from instrumenting our production files. I did test it by adding an unused generator function to `index.js` and in one of the `lib` files. That caused syntax errors in Node 0.10

**Comment by jamestalmage** [comment]
@urish Ready for your review and a merge

File: index.js (hunk 1 above)
Comment (urish) [inline index.js]: Great! I was wondering if there was a way to get rid of those errors

**Pull request merged**